cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/impl/CassandraSequenceHandler.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.cassandra.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.exceptions.DriverException;
+import com.datastax.driver.core.querybuilder.Insert;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+
+import org.hibernate.ogm.dialect.spi.NextValueRequest;
+import org.hibernate.ogm.model.key.spi.IdSourceKey;
+
+import static com.datastax.driver.core.querybuilder.QueryBuilder.eq;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.insertInto;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.select;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.set;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.update;
+
+/**
+ * Abstraction layer for adding id sequence numbers support to Cassandra,
+ * which lacks native auto-increment columns or sequences.
+ *
+ * @author Jonathan Halliday
+ */
+public class CassandraSequenceHandler {
+
+	private final CassandraDatastoreProvider provider;
+
+	public CassandraSequenceHandler(CassandraDatastoreProvider provider) {
+		this.provider = provider;
+	}
+
+	public void createSequence(String generatorsKey, CassandraDatastoreProvider datastoreProvider) {
+
+		// ogm wiring doesn't allow us to get at the real col names :-(
```

Comment:
The full info is indeed available at table creation time, but since it's not available at query time (i.e. in the NextValueRequest) we're left with the options of a) keeping it in some lookup structure on the sequencehandler class or b) just using hardcoded values that we know a-priori. The current impl is the latter option. let me know if you want to switch it.